apps/api/openapi/openapi.json

```diff
@@ -6009,6 +6009,178 @@
         }
       }
     },
+    "/api/user/competitions": {
+      "get": {
+        "summary": "Get competitions for user's agents",
+        "description": "Retrieve all competitions that the authenticated user's agents are participating in",
+        "tags": [
+          "User"
+        ],
+        "security": [
+          {
+            "SIWESession": []
+          }
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "status",
+            "schema": {
+              "type": "string",
+              "enum": [
+                "upcoming",
+                "active",
+                "ended"
+              ]
+            },
+            "description": "Filter competitions by status"
+          },
+          {
+            "in": "query",
+            "name": "claimed",
+            "schema": {
+              "type": "boolean"
+            },
+            "description": "Filter by claimed rewards status (not implemented)"
+          },
+          {
+            "in": "query",
+            "name": "limit",
+            "schema": {
+              "type": "integer",
+              "minimum": 1,
+              "maximum": 100,
+              "default": 10
+            },
+            "description": "Number of competitions to return"
+          },
+          {
+            "in": "query",
+            "name": "offset",
+            "schema": {
+              "type": "integer",
+              "minimum": 0,
+              "default": 0
+            },
+            "description": "Number of competitions to skip"
+          },
+          {
+            "in": "query",
+            "name": "sort",
+            "schema": {
+              "type": "string"
+            },
+            "description": "Sort order (e.g., \"startDate:desc\", \"name:asc\")"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "User agent competitions retrieved successfully",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "success": {
+                      "type": "boolean",
+                      "example": true
+                    },
+                    "userId": {
+                      "type": "string",
+                      "format": "uuid"
+                    },
+                    "competitions": {
```

Comment:
i think this is missing the `agents` object in #497? (sorry, i shouldve spec'd out the actual object instead of using bullet points)

i.e., the `competitions` array should competition objects which also include an `agents` array, and _this_ array should include which of the user's agents are in that specific competition
![image](https://github.com/user-attachments/assets/700b65c2-7e4e-4c41-b002-426a622dff32)

Reply:
```json
{
  "competitions": [
    {
      "id": "...",
      "name": "Example Competition",
      "status": "...",
      // can include other `Competition` fields, although UI doesnt require it
      "agents": [
        {
          "id": "...",
          "name": "Example Agent",
          "description": "Short description of what the agent does.",
          "walletAddress": "0xabc123...def",
          "rank": 1234
        }
      ]
    }
  ]
}
```

Reply:
Ah, I see.  So the `competitions.agents` array is filtered to only contain agents owned by the user making the request?

Reply:
I added agents, which includes rank.  You'll see the new logic for calculating rank below.